worker/Dockerfile

```diff
@@ -4,12 +4,18 @@ FROM node:12-alpine
 RUN apk update && apk upgrade && apk add --no-cache --virtual builds-deps build-base \
   g++ make python tini
 
-WORKDIR /usr/home/postmangovsg/
-COPY ./package* ./
+WORKDIR /usr/home/postmangovsg/worker
+
+COPY modules /usr/home/postmangovsg/modules
+
+RUN cd /usr/home/postmangovsg/modules/postman-templating && npm install && npm run build
```

Comment:
@lamkeewei why do we still need to install and build here, if the build runs on postinstall?

Reply:
Maybe we don't have to, I added this previously. Let me check if it is redundant.

Reply:
Hmmm so it seems like the original
`RUN cd /usr/home/postmangovsg/modules/postman-templating && npm install && npm run build`
Can be changed to 
`RUN cd /usr/home/postmangovsg/modules/postman-templating && npm install`

When we build on travis it seems like the `node_modules` in `modules/postman-templating` isn't copied into docker. 

As a result, when I try to run `npm ci` on `backend` in `Dockerfile`, it throws an error saying that `rimraf: not found` in `postman-templating`. 

Running `npm install` instead doesn't throw the error. 

I found this on stackoverflow: 
`npm ci (named after Continuous Integration) installs dependencies directly from package-lock.json and uses package.json only to validate that there are no mismatched versions. If any dependencies are missing or have incompatible versions, it will throw an error.`

https://travis-ci.com/github/opengovsg/postmangovsg/jobs/350810755

Do let me know if i got anything wrong; still not 100% sure what I am talking about

Reply:
I tried this out and is facing the same issue too. `npm install` works fine while `npm ci` gives the same error about `rimraf`. Probably got to find out more about the difference between `npm ci` and `npm install`.

On a separate note, I was able to drop the entire line: `RUN cd /usr/home/postmangovsg/modules/postman-templating && npm install && npm run build` and just do the`npm install` in `/usr/home/postmangovsg/workers`. 

My worker Dockerfile looks like this: 

```dockerfile
FROM node:12-alpine

# required for node-gyp
RUN apk update && apk upgrade && apk add --no-cache --virtual builds-deps build-base \
  g++ make python tini

WORKDIR /usr/home/postmangovsg/worker

COPY modules /usr/home/postmangovsg/modules
COPY worker/package* ./
RUN npm install

COPY worker/src ./src
COPY worker/tsconfig.json ./
RUN npm run build
RUN npm prune --production

EXPOSE 4000
ENTRYPOINT [ "tini", "--" ]
CMD ["npm", "start"]
```

You can test out the build locally by running `docker build -t worker -f worker/Dockerfile .` from the project root. If you have errors about missing modules, you can try removing `postman-templating/package-lock.json` and regenerating it with `npm install`.

Reply:
Hmmm so I was talking to chinying and he says that `npm ci` is preferred because `npm install` will install the `node_modules` according to the `package.json` and the version of the packages might change unexpectedly.

`npm ci` installs according to the `package-lock.json` so we know that whatever that is installed locally is what is installed on production.